AsyncDisplayKit/ASTextNode.mm

```diff
@@ -337,7 +337,18 @@ - (CGSize)calculateSizeThatFits:(CGSize)constrainedSize
 
   [self setNeedsDisplay];
   
-  return [[self _renderer] size];
+  CGSize size = [[self _renderer] size];
+  if (self.attributedString.length > 0) {
+    CGFloat screenScale = ASScreenScale();
+    self.ascender = round([[_attributedString attribute:NSFontAttributeName atIndex:0 effectiveRange:NULL] ascender] * screenScale)/screenScale;
```

Comment:
This code affects all text nodes, even those that don't have scaling enabled or that aren't inside a baseline-aligned stack.  Although this method is often called off main, there are a lot of critical situations where it is called on main, such as when transitioning to a view controller (modal or nav) and need to layout text positions so content doesn't jump; or in Pinterest, both opening a closeup as well as at the moment of dragging down to relayout the feed.

Is there any other code that uses these values besides the baseline stacks?  What do you think about creating accessor methods that lazily set the local property, or perhaps don't even cache it but just do the attribute / font accesses when requested?